src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -102,6 +110,8 @@ internal class PackageRestoreDataSource : ChainedProjectValueDataSourceBase<Rest
 
         internal async Task<IEnumerable<IProjectVersionedValue<RestoreData>>> RestoreAsync(IProjectVersionedValue<PackageRestoreUnconfiguredInput> e)
         {
+            await checkPendingNominationAsync(e);
+
             // No configurations - likely during project close
             if (!_enabled || e.Value.RestoreInfo is null)
```

Comment:
i guess we need add extra logic to skip the restore, if the input version has already out of dated.  But I guess we don't need do that within this PR, and can do it in a future PR (which we might also want to address multiple target framework bug I opened.